client-app/src/Components/DonorList.tsx

```diff
@@ -0,0 +1,201 @@
+        }
+    };
+
+    const handleAddNewDonationClick = () => {
+        navigate('/donorform');
+    };
+
+    const handleViewDetailsClick = () => {
+        navigate('/item');
+    };
+
+    return (
+        <>
+        <div>
+                <div className="header">
+                    <div className="logo-container">
+                        <img
+                            src="https://www.bworks.org/wp-content/themes/bworks/library/images/logo-bworks.png"
+                            alt="BWorks Logo"
+                            className="logo"
+                        />
+                    </div>
+                    <div
+                        style={{
+                            display: 'flex',
+                            alignItems: 'center',
+                            marginBottom: '10px',
+                        }}
+                    ></div>
+                    <div className="options">
+                        <div className="search-bar">
+                            <input
+                                type="text"
+                                placeholder="Search for donor"
+                                value={searchInput}
+                                onChange={e => setSearchInput(e.target.value)}
+                            />
+                            <button
+                                className="search-button"
+                                onClick={handleSearch}
+                            >
+                                <FaSearch />
+                            </button>
+                        </div>
+                    </div>
+                </div>
+
+                <table className="item-list">
+                    <thead>
+                        <tr>
+                            <th>Donor Number</th>
+                            <th>First Name</th>
+                            <th>Last Name</th>
+                            <th>Email</th>
+                            <th>More Details</th>
+                            {assignProgramClicked && <th>Select</th>}
+                        </tr>
+                    </thead>
+                    <tbody>
+                        {(filteredItems.length > 0
+                            ? filteredItems
+                            : donatedItems
+                        ).map((item, index) => (
+                            <tr key={item.donorId}>
+                                <td>{item.donorId}</td>
+                                <td>{item.firstName}</td>
+                                <td>{item.lastName}</td>
+                                <td>{item.email}</td>
+                                <td>
+                                    <button onClick={() => handleViewDetailsClick()}>
+                                        View More Details
+                                     </button>
+                                </td>
+
+                                {assignProgramClicked && (
```

Comment:
I see that you added `assignProgramClicked` in couple of places in this page. What is this for?